plugin/tls/tls.go

```diff
@@ -25,6 +26,24 @@ func setup(c *caddy.Controller) error {
 	return nil
 }
 
+func hardenTLS(tls *ctls.Config) {
```

Comment:
It's not really harden, it's settting some defaults that we think are good.

setTLSParameters, or just setTLSDefaults?